tests/admin_widgets/tests.py

```diff
@@ -639,9 +640,12 @@ def test_readonly_fields(self):
             html=True,
         )
         response = self.client.get(reverse("admin:admin_widgets_album_add"))
+        empty_value_display = AlbumAdmin(
+            Album, widget_admin_site
+        ).get_empty_value_display()
         self.assertContains(
             response,
-            '<div class="readonly"></div>',
+            '<div class="readonly">%s</div>' % empty_value_display,
```

Comment:
I'd add an empty value directly to the expected HTML.
```suggestion
            '<div class="readonly">-</div>',
```

Reply:
I guess the default value '-' for `empty_value_display` will never change in the future, so it's fair to use it literally.